internal/pkg/docker/dockercompose/decompose.go

```diff
@@ -0,0 +1,113 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package dockercompose
+
+import (
+	"fmt"
+	"github.com/aws/copilot-cli/internal/pkg/manifest"
+	"github.com/compose-spec/compose-go/loader"
+	compose "github.com/compose-spec/compose-go/types"
+	"sort"
+)
+
+// DecomposeService parses a Compose YAML file and then converts a single service to a Copilot manifest.
+func DecomposeService(content []byte, svcName string, workingDir string) (*manifest.BackendServiceConfig, IgnoredKeys, error) {
```

Comment:
Why is this `*manifest.BackendServiceConfig`? How about load balanced web service type?

Reply:
This is mostly just a temporary thing - since the code for Backend Service and Load Balanced Web Service is *almost* exactly the same (and I intend to use both), I'm just handling Backend Service now just to avoid lots of manual error-prone copy pasting of any updates to the code dealing with backend service / LBWS.

Though, this is a good point - since this is likely going to be the public-facing method of the conversion, it should be able to return an LBWS as well no matter what.

I'm still thinking about what the best way to return one or the other from this function, though, and the signature of this function could certainly be updated in this PR. Maybe this signature would be the best? Or is the number of return values starting to get excessive here?

```golang
func DecomposeService(content []byte, svcName string, workingDir string) (*manifest.BackendServiceConfig, *manifest.LoadBalancedWebServiceConfig, IgnoredKeys, error) {
```

Reply:
Maybe we can use go generic to avoid duplicated code blocks. One example would be how we deal with [`AdvancedScalingConfig`](https://github.com/aws/copilot-cli/blob/7980a515b98156dfbce4c3070edc101061b00a0e/internal/pkg/manifest/svc.go#L206).

Reply:
I didn't even think of generics! I'll take a look 👀

Reply:
Unfortunately, it seems like this won't allow us to eliminate the main classes of boilerplate because we can't access the shared struct fields: https://stackoverflow.com/a/71531151

Reply:
Actually why do we need to return a concrete type like `*manifest.BackendServiceConfig`? If we'll output a manifest anyway can we just return an interface with a write function that output locally? Something like `encoding.BinaryMarshaler`?

Reply:
That's a fair point! I'd have to investigate how to handle the testing for that. My main concern would be that it might go against the principle of returning as concrete of a type as possible.

Reply:
Hm, yeah...especially since we don't yet know how/where this will get used, a write function makes more sense than using one of our concrete types.